core/modules/catalog/store/product/actions.ts

```diff
@@ -686,6 +686,7 @@ const actions: ActionTree<ProductState, RootState> = {
   /**
    * Load the product data - async version for asyncData()
    */
```

Comment:
Plus - we need to simplify the asyncData/validateRoute calls to this method to simplify it as it was done on the category page 

If we add references to category-next they could be done just from Product.vue in case of somebody using Product.js (backward compatibility mode) to not add any reference to category-next in this case

I like this approach, let’s go this way.

Do you plan to add some simplified / refactored version of product/list (maybe decoupled into set of 2-3 helpers) and product/single by this same refactoring effort? Or just do this as a next step? (This staging approach makes perfect sense to me)